test/async_scope_v1_test.cpp

```diff
@@ -0,0 +1,661 @@
+            },
+            [&scheduled, &finished](auto&) noexcept {
+              return sequence(
+                  just_from([&scheduled]() noexcept { scheduled.set(); }),
+                  finished.async_wait());
+            });
+      }));
+
+  // ensure the spawned work is actually spawned before...
+  sync_wait(scheduled.async_wait());
+
+  // ...dropping the future
+  optFuture.reset();
+
+  // we know that the stop callback has been registered (that happens before the
+  // spawned work sets the scheduled event) so dropping the future ought to
+  // trigger the callback and set wasStopped to true
+  EXPECT_TRUE(wasStopped.load());
+
+  // now clean up the test state; release the awaited event and block until the
+  // scope sees the work finish (skipping this last step causes a race between
+  // waking up the blocked work and destroying finished)
+
+  finished.set();
+
+  sync_wait(scope.complete());
+}
+
+TEST_F(async_scope_test, requesting_the_scope_stop_cancels_pending_futures) {
+  async_manual_reset_event evt;
+
+  auto fut = scope.spawn_on(thread.get_scheduler(), evt.async_wait());
+
+  scope.request_stop();
+
+  // with the scope cancelled, pending futures should all immediately complete
+  // with done
+  auto result = sync_wait(std::move(fut));
+
+  EXPECT_FALSE(result);
+
+  // but the scope itself won't complete until the spawned work is actually done
+  // so we need to release the event here and block on scope completion before
+  // the event is destroyed to make sure the test actually completes
+  evt.set();
+
+  sync_wait(scope.complete());
+}
+
+TEST_F(async_scope_test, spawning_after_cleaning_up_destroys_the_sender) {
+  spawn_work_after_cleanup();
+}
+
+TEST_F(async_scope_test, cleanup_is_idempotent) {
+  sync_wait(scope.cleanup());
+
+  spawn_work_after_cleanup();
+}
+
+TEST_F(async_scope_test, spawning_work_makes_it_run) {
+  expect_work_to_run();
+
+  sync_wait(scope.cleanup());
+}
+
+TEST_F(async_scope_test, spawning_work_makes_it_run_with_lambda) {
+  expect_work_to_run_call_on();
+
+  sync_wait(scope.cleanup());
+}
+
+TEST_F(async_scope_test, scope_not_stopped_until_cleanup_is_started) {
+  auto cleanup = scope.cleanup();
+
+  expect_work_to_run();
+
+  sync_wait(std::move(cleanup));
+}
+
+TEST_F(async_scope_test, work_spawned_in_correct_context) {
+  auto futureId = scope.spawn_on(thread.get_scheduler(), just_from([] {
+                                   return std::this_thread::get_id();
+                                 }));
+  auto id = sync_wait(std::move(futureId));
+  sync_wait(scope.cleanup());
+  ASSERT_TRUE(id);
+  EXPECT_EQ(*id, thread.get_thread_id());
+  EXPECT_NE(*id, std::this_thread::get_id());
+}
+
+TEST_F(async_scope_test, lots_of_threads_works) {
+#if !defined(UNIFEX_TEST_LIMIT_ASYNC_SCOPE_THREADS)
+  constexpr int maxCount = 1'000;
+#else
+  constexpr int maxCount = UNIFEX_TEST_LIMIT_ASYNC_SCOPE_THREADS;
+#endif
```

Comment:
This conditional compilation was added to support an internal project that has since been killed.  Maybe we can simplify.